src/Agent.Worker/WorkerCommandManager.cs

```diff
@@ -160,20 +160,19 @@ protected void InstallWorkerCommand(IWorkerCommand commandExecutor)
         {
             if (_commands.ContainsKey(commandExecutor.Name))
             {
-                //TODO: Add some logging (or throw exception)
-                return;
+                throw new Exception(StringUtil.Loc("CommandDuplicateDetected", commandExecutor.Name, CommandArea.ToLowerInvariant()));
             }
             _commands[commandExecutor.Name] = commandExecutor;
             var aliasList = commandExecutor.Aliases;
             if (aliasList != null)
             {
                 foreach (var alias in commandExecutor.Aliases)
                 {
-                    if (!_commands.ContainsKey(alias))
+                    if (_commands.ContainsKey(alias))
                     {
-                        _commands[alias] = commandExecutor;
+                        throw new Exception(StringUtil.Loc("CommandDuplicateDetected", alias, CommandArea.ToLowerInvariant()));
                     }
-                    // TODO: else add some logging (or throw exception)
```

Comment:
Previously this would fail silently and now we are going to blow. How sure are we that won't happen? :)

Reply:
If it blows up, we know we didn't do a conversion correctly and we are glad we caught it before it went out :)